### ISSUE TITLE
Integrate debugging

### ISSUE THREAD

**Issue by basarat** [TypeStrong/atom-typescript#316]
Don't make the UI. Make it easier to remote control `node-inspector` and Chrome dev tools
## Existing packages

https://atom.io/packages/node-debugger
https://github.com/mark-hahn/node-ide
### Chrome debugger protocol:
- https://developer.chrome.com/devtools/docs/debugger-protocol

Specifically for debugger: https://developer.chrome.com/devtools/docs/protocol/1.1/debugger
### Tutorial

Run chrome in debug mode

``` ts
chrome.exe --remote-debugging-port=9222 --user-data-dir=c:\chromedev
```

The `user-data-dir` is just a trick to make it a _completely new_ chrome instance. Otherwise if you already have chrome running the new chrome _sub process_ will not start listening. If you don't have any chrome instances running you don't need this flag.

Connect to it from another chrome by visiting

``` ts
chrome.exe http://localhost:9222
```
### Package

Just use : https://www.npmjs.com/package/chrome-remote-interface featured on https://developer.chrome.com/devtools/docs/debugging-clients
## Launch config

Here is a sample launch configuration from VS code `.settings/launch.json`: 

```
{
    "version": "0.1.0",
    // List of configurations. Add new configurations or edit existing ones.  
    // ONLY "node" and "mono" are supported, change "type" to switch.
    "configurations": [
        {
            // Name of configuration; appears in the launch configuration drop down menu.
            "name": "Launch app.js",
            // Type of configuration. Possible values: "node", "mono".
            "type": "node",
            // Workspace relative or absolute path to the program.
            "program": "app.js",
            // Automatically stop program after launch.
            "stopOnEntry": true,
            // Command line arguments passed to the program.
            "args": [],
            // Workspace relative or absolute path to the working directory of the program being debugged. Default is the current workspace.
            "cwd": ".",
            // Workspace relative or absolute path to the runtime executable to be used. Default is the runtime executable on the PATH.
            "runtimeExecutable": null,
            // Environment variables passed to the program.
            "env": { }
        }, 
        {
            "name": "Attach",
            "type": "node",
            // TCP/IP address. Default is "localhost".
            "address": "localhost",
            // Port to attach to.
            "port": 5858
        }
    ]
}
```

**Comment by tinganho** [comment]
Can't wait for this :+1:

**Comment by basarat** [comment]
@tinganho chrome or node-inspector?

**Comment by tinganho** [comment]
Node-inspector. It would be awesome if everything could be controlled in the Atom, even when debugging a web app.

**Comment by Roaders** [comment]
Yeah, this sounds great. I want chrome and Node-inspector (I'm greedy)

**Comment by jhabdas** [comment]
Hello from the future...

Related thread to add source maps support for Atom's Node Debugger:
https://github.com/kiddkai/atom-node-debugger/issues/29#issuecomment-264864660

A reference implementation from MS for VSCode:
https://github.com/Microsoft/vscode-node-debug2

Using TypeStrong's own `ts-node` for experimental debugging in VSCode:
https://github.com/Enterprise-JS/vscode-ts-node-debugging
